Makefile

```diff
@@ -164,6 +166,14 @@ generate: controller-gen
 	go generate ./...
 	$(CONTROLLER_GEN) object:headerFile="hack/boilerplate.go.txt" paths="./..."
 
+# Check and add (if missing) license header
+LICENSE_FILE=$(shell pwd)/LICENSE
+.PHONY: add-license 
+add-license: addlicense
+	echo "Checking/Adding license..."
+	$(eval GO_PACKAGES  := $(shell go list -f '{{.Dir}}' ./... | sed -e "s|^$$(pwd)||" | cut -d/ -f2 | uniq))
```

Comment:
I think it's fine to declare this outside of the recipe, and remove the `eval` part. This should make it a little simpler.

Reply:
There should still be a `sort` in the pipeline before `uniq`, because `uniq` only detects repeated lines if they are adjacent to each other.

Or you can use `sort -u` instead of `sort | uniq`

Reply:
Unless `go list` guarantees the output ordering will already be sorted, though adding `sort` to the pipeline is really low impact in the redundant case anyway.

Reply:
`go help list` also explains:

```
The -find flag causes list to identify the named packages but not
resolve their dependencies: the Imports and Deps lists will be empty.

The -test flag causes list to report not only the named packages
but also their test binaries (for packages with tests), to convey to
source code analysis tools exactly how test binaries are constructed.
The reported import path for a test binary is the import path of
the package followed by a ".test" suffix, as in "math/rand.test".
When building a test, it is sometimes necessary to rebuild certain
dependencies specially for that test (most commonly the tested
package itself). The reported import path of a package recompiled
for a particular test binary is followed by a space and the name of
the test binary in brackets, as in "math/rand [math/rand.test]"
or "regexp [sort.test]". The ForTest field is also set to the name
of the package being tested ("math/rand" or "sort" in the previous
examples).
```

Adding `-find` speeds up the invocation.

```bash
$ time go list -f '{{.Dir}}' ./...
/home/andrew/workspace/cryostat-operator/api/v1beta1
/home/andrew/workspace/cryostat-operator/internal
/home/andrew/workspace/cryostat-operator/internal/controllers
/home/andrew/workspace/cryostat-operator/internal/controllers/client
/home/andrew/workspace/cryostat-operator/internal/controllers/common
/home/andrew/workspace/cryostat-operator/internal/controllers/common/resource_definitions
/home/andrew/workspace/cryostat-operator/internal/test

real	0m0.242s
user	0m0.906s
sys	0m0.223s
```

```bash
$ time go list -find -f '{{.Dir}}' ./...
/home/andrew/workspace/cryostat-operator/api/v1beta1
/home/andrew/workspace/cryostat-operator/internal
/home/andrew/workspace/cryostat-operator/internal/controllers
/home/andrew/workspace/cryostat-operator/internal/controllers/client
/home/andrew/workspace/cryostat-operator/internal/controllers/common
/home/andrew/workspace/cryostat-operator/internal/controllers/common/resource_definitions
/home/andrew/workspace/cryostat-operator/internal/test

real	0m0.117s
user	0m0.344s
sys	0m0.130s
```

Using `-test` finds some more (duplicated) source directories:

```bash
$ time go list -test -f '{{.Dir}}' ./...
/home/andrew/workspace/cryostat-operator/api/v1beta1
/home/andrew/workspace/cryostat-operator/internal
/home/andrew/workspace/cryostat-operator/internal/controllers
/home/andrew/workspace/cryostat-operator/internal/controllers/client
/home/andrew/workspace/cryostat-operator/internal/controllers/common
/home/andrew/workspace/cryostat-operator/internal/controllers/common/resource_definitions
/home/andrew/workspace/cryostat-operator/internal/test
/home/andrew/workspace/cryostat-operator/internal/controllers
/home/andrew/workspace/cryostat-operator/internal/controllers

real	0m0.305s
user	0m1.004s
sys	0m0.256s
```

These flags can't be used together, though.

I think using `-test` with a `sort | uniq` in the pipeline is probably the most correct and complete listing.

Reply:
Cool thanks @andrewazores Just added the changes!